### PR TITLE
Update config and run Detekt during the build

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
This changeset updates `config` and includes Detekt code analysis into the build.

Apart from that, it does the following:

1. Employs `Spine` dependency object to handle versions of Spine module instead of `extras` in `version.gradle.kts`.
2. Adds `pull` script to the root directory.
3. Moves forcing of Dokka into `buildSrc`.

Dokka is widely used across our modules now and usually forced, so I suggest moving its forcing into `config` under `forceProductionDependencies()` method.

The version is bumped to `2.0.0-SNAPSHOT.2`.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -44,30 +44,40 @@ all modules and does not describe the project structure per-subproject.
...
   </dependency>
   <dependency>
     <groupId>com.puppycrawl.tools</groupId>
     <artifactId>checkstyle</artifactId>
     <version>10.3.4</version>
   </dependency>
+  <dependency>
+    <groupId>io.gitlab.arturbosch.detekt</groupId>
+    <artifactId>detekt-cli</artifactId>
+    <version>1.21.0</version>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:79]: ignore

File: pom.xml [hunk 2]
```diff
@@ -44,30 +44,40 @@ all modules and does not describe the project structure per-subproject.
...
     <artifactId>junit-jupiter-engine</artifactId>
-    <version>5.8.2</version>
+    <version>5.9.1</version>
     <scope>test</scope>
   </dependency>
   <dependency>
     <groupId>com.google.errorprone</groupId>
     <artifactId>error_prone_core</artifactId>
-    <version>2.13.1</version>
+    <version>2.16</version>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:59]: ignore

[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:59]: It's already the latest version.

**Comment by yevhenii-nadtochii** [comment]
Also, this module emits the following warning:

```
The project uses several versions of `io.spine.validation:spine-validation-java-runtime` dependency.
module: text, configuration: implementation, version: 2.0.0-SNAPSHOT.61
module: text, configuration: implementation, version: 2.0.0-SNAPSHOT.40
```

Looks like `DependencyWriter` examines unresolved version of `implementation` configuration. This library is forced directly in the build script. And, as far as I understand, the actually used version should go to `POM` report. 

Screenshot from the build scan:

image

Can I create an issue in `config`?

**Comment by yevhenii-nadtochii** [comment]
@alexander-yevsyukov @armiol Please take a look on PR itself and a comment I left.

**Comment by armiol** [comment]
@yevhenii-nadtochii the real issue is that this library actually allows two versions of the same dependency. It means we don't fail on conflict. Which we should.

**Comment by armiol** [comment]
@yevhenii-nadtochii other than that, it's quite normal to have a couple of versions of the same dependency, one of which is forced.

**Comment by armiol** [comment]
@yevhenii-nadtochii and yes, we should have the one actually used in our POM. I would first address it in this repo to verify the check works properly.

**Review by armiol — APPROVED** [review]

**Review by alexander-yevsyukov — APPROVED** [review]
LGTM

**Pull request merged**